server/member/global_tso_allocator.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2020 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package member
```

Comment:
Why is TSO related logic in member package?

Reply:
After adding the new feature of local TSO allocator, there will be two kinds of member roles need to be elected, one is for PD leader, another is for TSO Allocator. So I redefine or regard `member` package as a place to implement **the member roles that need to be elected**. For more details, you can check this [document](https://docs.google.com/document/d/1t4D2M5W0dqduNTey1aTute4umuEQrKazU-ipSwFAaLM/edit).

Reply:
I added some explanations in the description. PTAL. : -)

Reply:
why cannot use `TSOAllocater` as a common interface in `tso` package, and also implement Local and Global tso allocator in  `tso` package?